### PR TITLE
Add support for RUN --network=none --security=insecure

### PR DESCRIPTION
**Pull request by rhatdan**
Docker now supports --network and --security option on the RUN line in Dockerfiles, this PR adds support for both.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
@flouthoc @nalind PTAL

**Comment by nalind** [comment]
The `dockerclient` implementation needs to complain about features it doesn't support.

**Comment by rhatdan** [comment]
@nalind not sure where that happens?

**Comment by nalind** [comment]
`dockerclient.ClientExecutor.Run()`.  Places where we've neglected this before produced #187, which was a bad experience for users who run the `imagebuilder` binary.

File: builder.go [hunk 1]
```diff
@@ -422,6 +426,10 @@ func (b *Builder) Run(step *Step, exec Executor, noRunsRemaining bool) error {
 	for _, run := range runs {
 		config := b.Config()
 		config.Env = step.Env
+		config.NetworkDisabled = (run.Network == "none")
```
Comment (flouthoc) [inline builder.go:429]: Shouldn't `config.NetworkDisabled` must be set on implementation end i.e buildah and imagebuilder should only parse `--network` and just set value in `run.Network` ? I always think of imagebuilder as the the frontend and buildah/docker as the actual implementation backend.

@nalind Could you correct if i got this wrong.

Comment (rhatdan) [reply builder.go:429]: The only field to pass information to buildah is the NetworkDisabled field.  Even in the --security=insecure, there is no Privileged field to set, so I use the SecurityOpts to pass it through.

Comment (nalind) [reply builder.go:429]: `imagebuilder` provides the parser in `dockerfile`, a builder in its top level package, and an executor that talks to an engine API (by way of go-dockerclient) that the builder can use in its `dockerclient` package.  Buildah uses the first two and supplies its own executor.

**Comment by rhatdan** [comment]
@flouthoc Please take this over.  I am not going to have time to complete it

**Comment by nalind** [comment]
I would also suggest dropping `--security`.  It's a "labs" feature in BuildKit, and not enabled there by default (compared to the current version of this PR, which makes it available all the time), and we don't have an identified use case for it.  What's the benefit to `imagebuilder`?

**Comment by rhatdan** [comment]
The only use case would be if you wanted to run a --privileged container from within a build.  But I am fine with dropping it.

**Comment by flouthoc** [comment]
I'm taking over this PR and will create two different PR for `RUN --network` and `RUN --security`

**Comment by nalind** [comment]
I would again suggest dropping `--security`.

**Comment by flouthoc** [comment]
> I would again suggest dropping `--security`.

@nalind Ackd dropping this for now.

**Comment by nalind** [comment]
What are the use case and argument for overriding the specified or default value of the `--network` CLI setting from inside of a Dockerfile?  BuildKit at least requires that the user decide to allow these things at build-time, and I don't see any such precaution in here.

**Comment by flouthoc** [comment]
> What are the use case and argument for overriding the specified or default value of the `--network` CLI setting from inside of a Dockerfile? BuildKit at least requires that the user decide to allow these things at build-time, and I don't see any such precaution in here.

I haven't check the PR implementation yet but I'd assume that `RUN --network` should only override for a specific `RUN` invocation and not for the entire build. So for instance a build can be invoked with CLI `--network=host` but certain `RUN` instructions can be isolated with `RUN --network=none <some task>`

**Comment by rhatdan** [comment]
Yes that is what I would expect.

**Comment by rhatdan** [comment]
When I originally looked at this, it looked like imagebuilder and buildah were setup to allow the override on a per RUN bases.

I guess if you were overly cautious you might have

RUN dnf -y update ...
RUN --network=none /run/local/script

**Comment by flouthoc** [comment]
Should we close this one in favor of: https://github.com/openshift/imagebuilder/pull/247 and https://github.com/containers/buildah/pull/4566

**Comment by rhatdan** [comment]
Sure.

**Pull request closed without merging**